File: HabitFormationApp/src/HabitFormationApp.Core/Model/ReminderAggregate/Reminder.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace HabitFormationApp.Core;

public class Reminder
{
    public Guid ReminderId { get; set; }
    public Guid UserId { get; set; }

    /// <summary>
    /// Gets or sets the tenant ID for multi-tenant isolation.
    /// </summary>
    public Guid TenantId { get; set; }
    public Guid HabitId { get; set; }
    public TimeSpan ReminderTime { get; set; }
    public string? Message { get; set; }
    public bool IsEnabled { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    public void Toggle()
    {
        IsEnabled = !IsEnabled;
    }
}

File: HabitFormationApp/src/HabitFormationApp.WebApp/projects/habit-formation-app/src/app/models/reminder.ts
export interface Reminder {
  reminderId: string;
  userId: string;
  habitId: string;
  reminderTime: string;
  message?: string;
  isEnabled: boolean;
  createdAt: string;
}

export interface CreateReminderRequest {
  userId: string;
  habitId: string;
  reminderTime: string;
  message?: string;
  isEnabled: boolean;
}

export interface UpdateReminderRequest {
  reminderId: string;
  reminderTime: string;
  message?: string;
  isEnabled: boolean;
}
